components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1691,13 +1691,15 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
 
         const teamProjects = await this.projectsService.getTeamProjects(teamId);
         teamProjects.forEach(project => {
-            this.deleteProject(ctx, project.id);
-        })
+            /** no awat */ this.deleteProject(ctx, project.id)
```

Comment:
Typo

```suggestion
            /** no await */ this.deleteProject(ctx, project.id)
```

Reply:
Actually, shouldn't we `await` here? 🤔

```ts
        const teamProjects = await this.projectsService.getTeamProjects(teamId);
        await Promise.all(teamProjects.map(project => this.deleteProject(ctx, project.id));
```

Reply:
don't think so. why blocking the deletion of a team?

Reply:
Because when you delete a team, you want to be sure everything is gone, right? I.e. if this breaks mid-way due to a bug, maybe it's preferable to get an error and reach out to support for full deletion, as opposed to leaving potentially sensitive data behind? 💭

But maybe I'm being too paranoid here and deleting async / in the background is actually fine. 😊

Reply:
We had the very same situation for user accounts and this would block re-creating team as well.